calcom/cal.com
# fix: the 500 error when inviting member to team via link

## What does this PR do?


Inviting someone to the team via link was throwing a 500 error. The error was due to the use of `useSuspenseQuery` in the `teams` page. When the API throws `UNAUTHORIZED` error the suspense also throws an error. I changed it to `useQuery`, and this fixed the propagation of the error.

Fixes #10906 

Video after the fix. The incognito window demonstrates the link opening when the user is not logged in. 

[Video](https://github.com/calcom/cal.com/assets/55000107/70fadf2d-8d52-4bb9-9eec-bbacf8a4b1b7)

## Type of change

- [ x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?


To test the solution, generate a team invite link and try to open it after logging out of the account or from the incognito window. Earlier it used to throw error. Now it should redirect to the login page and after loging-in the team invite will be shown.

## Mandatory Tasks

- [ x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.
